### DISCUSSION TITLE
`docanalysis` alpha-testing

### DISCUSSION THREAD

**Discussion by ShweataNHegde** [petermr/petermr#43] [General]
_Use this Discussion thread to document your experience running `docanalysis`_. 

Please open one comment for yourself and update as you go. If `docanalysis` crashes or if you have installation issues, please raise an Issue, here: https://github.com/petermr/docanalysis/issues. Thanks! 

Here's the tutorial to follow: https://github.com/petermr/docanalysis/wiki/docanalysis-Tutorial/

**Comment by ShweataNHegde** [comment]
Format to follow

### Tester info
- Tester name: 
- OS:
- Python version: 
---
### List of commands and outputs 
Document crashes; unexpected behaviour, etc.

**Comment by Pg-Pi3Gey** [comment]
### Tester info
Tester name: Piyush Gajbhiye
OS: Windows 10 64-bit Build 19041
Python version: Python 3.10.4

### List of commands and outputs
ran `pip install docanalysis` after following the setting up instruction from doc.
getting this error and docanalysis is not installing
```
 "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tpnmslib.cc /Fobuild\temp.win-amd64-3.10\Release\nmslib.obj /EHsc /openmp /O2 /DVERSION_INFO=\\\"2.1.1\\\"
      nmslib.cc
      C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\object.h(127): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
      C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\distcomp.h(260): warning C4244: 'initializing': conversion from 'size_t' to 'float', possible loss of data
      nmslib.cc(454): error C2017: illegal escape sequence
      nmslib.cc(454): error C2001: newline in constant
      nmslib.cc(459): error C2143: syntax error: missing ')' before 'pybind11::enum_<similarity::DistType>'
      nmslib.cc(459): error C2143: syntax error: missing ';' before 'pybind11::enum_<similarity::DistType>'
      nmslib.cc(730): warning C4267: '=': conversion from 'size_t' to 'T', possible loss of data
              with
              [
                  T=int
              ]
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> nmslib

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by petermr** [comment]
This is deep inside the build of some component (it's a compile error in a
C program). Something is probably the wrong version. You may have to update
a compiler.

Suggest looking on Google for this error. Else try updating compilers.

On Thu, Jul 14, 2022 at 10:44 AM Pg ***@***.***> wrote:

> Tester info
>
> Tester name: Piyush Gajbhiye
> OS: Windows 10 64-bit Build 19041
> Python version: Python 3.10.4
> List of commands and outputs
>
> ran pip install docanalysis after following the setting up instruction
> from doc.
> getting this error and docanalysis is not installing
>
>  "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tpnmslib.cc /Fobuild\temp.win-amd64-3.10\Release\nmslib.obj /EHsc /openmp /O2 /DVERSION_INFO=\\\"2.1.1\\\"
>
>       nmslib.cc
>
>       C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\object.h(127): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
>
>       C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\distcomp.h(260): warning C4244: 'initializing': conversion from 'size_t' to 'float', possible loss of data
>
>       nmslib.cc(454): error C2017: illegal escape sequence
>
>       nmslib.cc(454): error C2001: newline in constant
>
>       nmslib.cc(459): error C2143: syntax error: missing ')' before 'pybind11::enum_<similarity::DistType>'
>
>       nmslib.cc(459): error C2143: syntax error: missing ';' before 'pybind11::enum_<similarity::DistType>'
>
>       nmslib.cc(730): warning C4267: '=': conversion from 'size_t' to 'T', possible loss of data
>
>               with
>
>               [
>
>                   T=int
>
>               ]
>
>       error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
>
>       [end of output]
>
>
>
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>
> error: legacy-install-failure
>
>
>
> × Encountered error while trying to install package.
>
> ╰─> nmslib
>
>
>
> note: This is an issue with the package mentioned above, not pip.
>
> hint: See above for output from the failure.
>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/43#discussioncomment-3145447>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS6HZJ2MX7AHC7Z2AYLVT7OOPANCNFSM53PB6ZGQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by petermr** [comment]
Looks like a Python 3.8 / 3.10 incompatibility.

See https://github.com/nmslib/nmslib/issues/464

try downgrading to Python 3.8 (which is what I run).
Thanks for your patience.

On Thu, Jul 14, 2022 at 9:18 PM Peter Murray-Rust <
***@***.***> wrote:

> This is deep inside the build of some component (it's a compile error in a
> C program). Something is probably the wrong version. You may have to update
> a compiler.
>
> Suggest looking on Google for this error. Else try updating compilers.
>
> On Thu, Jul 14, 2022 at 10:44 AM Pg ***@***.***> wrote:
>
>> Tester info
>>
>> Tester name: Piyush Gajbhiye
>> OS: Windows 10 64-bit Build 19041
>> Python version: Python 3.10.4
>> List of commands and outputs
>>
>> ran pip install docanalysis after following the setting up instruction
>> from doc.
>> getting this error and docanalysis is not installing
>>
>>  "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt" /EHsc /Tpnmslib.cc /Fobuild\temp.win-amd64-3.10\Release\nmslib.obj /EHsc /openmp /O2 /DVERSION_INFO=\\\"2.1.1\\\"
>>
>>       nmslib.cc
>>
>>       C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\object.h(127): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data
>>
>>       C:\Temp\pip-install-qii4qkv_\nmslib_8c50588dc520411fa08b71247222a315\similarity_search\include\distcomp.h(260): warning C4244: 'initializing': conversion from 'size_t' to 'float', possible loss of data
>>
>>       nmslib.cc(454): error C2017: illegal escape sequence
>>
>>       nmslib.cc(454): error C2001: newline in constant
>>
>>       nmslib.cc(459): error C2143: syntax error: missing ')' before 'pybind11::enum_<similarity::DistType>'
>>
>>       nmslib.cc(459): error C2143: syntax error: missing ';' before 'pybind11::enum_<similarity::DistType>'
>>
>>       nmslib.cc(730): warning C4267: '=': conversion from 'size_t' to 'T', possible loss of data
>>
>>               with
>>
>>               [
>>
>>                   T=int
>>
>>               ]
>>
>>       error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.32.31326\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
>>
>>       [end of output]
>>
>>
>>
>>   note: This error originates from a subprocess, and is likely not a problem with pip.
>>
>> error: legacy-install-failure
>>
>>
>>
>> × Encountered error while trying to install package.
>>
>> ╰─> nmslib
>>
>>
>>
>> note: This is an issue with the package mentioned above, not pip.
>>
>> hint: See above for output from the failure.
>>
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/petermr/petermr/discussions/43#discussioncomment-3145447>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAFTCS6HZJ2MX7AHC7Z2AYLVT7OOPANCNFSM53PB6ZGQ>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>
>
> --
> Peter Murray-Rust
> Founder ContentMine.org
> and
> Reader Emeritus in Molecular Informatics
> Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK
